web-ui/src/main/resources/catalog/components/metadataactions/RelatedDirective.js

```diff
@@ -178,6 +178,33 @@
                            scope.relations[idx] = value;
                          }
 
+                         // For draft version append "approved=false" to url
+                         if (scope.md.draft === "y" && idx === "onlines") {
+                           for (var i = 0; i < scope.relations[idx].length; i++) {
+                             if (scope.relations[idx][i].url) {
+                               if (typeof scope.relations[idx][i].url === 'string') {
+                                 if (
+                                   scope.relations[idx][i].url.match(
+                                     ".*/api/records/" + scope.md['geonet:info'].uuid + "/attachments/.*"
+                                   ) != null
+                                 ) {
+                                   scope.relations[idx][i].url += "?approved=false";
+                                 }
+                               } else if (typeof scope.relations[idx][i].url === 'object') {
```

Comment:
Should this be 

`} else if (Array.isArray(scope.relations[idx][i].url)) {`

Reply:
@ianwallen the url is a string (not very clear if when this happens) or an object containing the url related to the metadata main language, but not an array. 

Creating the a metadata with the default iso19139 template and wvaluating `scope.relations[idx][i].url`.

```
{eng: 'http://localhost:8080/geonetwork/srv/api/records/UUID/attachments/test-image.png'}
```